packaging/Microsoft.NET.Sdk.WindowsDesktop/targets/Microsoft.NET.Sdk.WindowsDesktop.props

```diff
@@ -34,16 +34,41 @@
     <FrameworkReference Include="Microsoft.WindowsDesktop.App.WindowsForms" IsImplicitlyDefined="true" />
   </ItemGroup>
 
+  <!--
+    Traditionally, Visual Studio has supplied these references for .NET Framework based
+    WPF Projects: 
+    
+    .NET 3.x:   PresentationCore, PresentationFramework, WindowsBase 
+    .NET 4.x:   PresentationCore, PresentationFramework, WindowsBase, System.Xaml 
+
+
+    Microsoft.NET.WindowsDesktop.SDK will supply the following references to .NET Framework based 
+    WPF Projects: 
+    
+    .NET 3.x:   PresentationCore, PresentationFramework, WindowsBase 
+    
+    .NET 4.0:   PresentationCore, PresentationFramework, WindowsBase, System.Xaml, 
+                UIAutomationClient, UIAutomationClientSideProviders, UIAutomationProvider, UIAutomationTypes
+                
+    .NET 4.5+:  PresentationCore, PresentationFramework, WindowsBase, System.Xaml, 
+                UIAutomationClient, UIAutomationClientSideProviders, UIAutomationProvider, UIAutomationTypes
+                System.Windows.Controls.Ribbon
+  --> 
   <ItemGroup Condition=" '$(DisableImplicitFrameworkReferences)' != 'true' and '$(TargetFrameworkIdentifier)' == '.NETFramework' And '$(UseWPF)' == 'true' ">
     <_SDKImplicitReference Include="WindowsBase"/>
     <_SDKImplicitReference Include="PresentationCore"/>
     <_SDKImplicitReference Include="PresentationFramework"/>
-    <_SDKImplicitReference Include="System.Windows.Controls.Ribbon"/>
-    <_SDKImplicitReference Include="System.Xaml"/>
+  </ItemGroup>
+
+  <ItemGroup Condition=" '$(DisableImplicitFrameworkReferences)' != 'true' and '$(TargetFrameworkIdentifier)' == '.NETFramework' And '$(UseWPF)' == 'true' And $(TargetFramework.StartsWith('net4'))">
+    <_SDKImplicitReference Include="System.Xaml">
+      <RequiredTargetFramework>4.0</RequiredTargetFramework>
+    </_SDKImplicitReference>
     <_SDKImplicitReference Include="UIAutomationClient"/>
     <_SDKImplicitReference Include="UIAutomationClientSideProviders"/>
     <_SDKImplicitReference Include="UIAutomationProvider"/>
     <_SDKImplicitReference Include="UIAutomationTypes"/>
+    <_SDKImplicitReference Condition="'$(TargetFramework)' != 'net40'" Include="System.Windows.Controls.Ribbon"/>
```

Comment:
We shouldn't use $(TargetFramework), it should be a user value like $(Configuration). TargetFramework can have more than one spelling. For example, net4.0 is also .NET 4.0.

This needs to check the $(TargetFrameworkVersion).

Reply:
You can also do more than exact comparisons when you strip off the `v`. We have the $(_TargetFrameworkVersionWithoutV) defined. Regrettable that this was marked "private", but others already use it and I am sure we'll have to support it forever. :)

See https://github.com/microsoft/msbuild/issues/3212, which covers some of the pitfalls of version comparisons in msbuild and discusses some paths forward that would make checking TargetFrameworkVersion less error prone. Unfortunately, it is quite tricky today. :(

Reply:
@nguerrera, resolved using `$(_TargetFrameworkVersionWithoutV)`